src/hotspot/share/gc/g1/g1PeriodicGCTask.cpp

```diff
@@ -54,11 +54,17 @@ bool G1PeriodicGCTask::should_start_periodic_gc(G1CollectedHeap* g1h,
 
   // Check if load is lower than max.
   double recent_load;
-  if ((G1PeriodicGCSystemLoadThreshold > 0.0f) &&
-      (os::loadavg(&recent_load, 1) == -1 || recent_load > G1PeriodicGCSystemLoadThreshold)) {
-    log_debug(gc, periodic)("Load %1.2f is higher than threshold %1.2f. Skipping.",
-                            recent_load, G1PeriodicGCSystemLoadThreshold);
-    return false;
+  if (G1PeriodicGCSystemLoadThreshold > 0.0) {
+    if (os::loadavg(&recent_load, 1) == -1) {
+      G1PeriodicGCSystemLoadThreshold = 0.0;
+      log_warning(gc, periodic)("System loadavg not supported, G1PeriodicGCSystemLoadThreshold check is disabled.");
+      return false;
+    }
+    if (recent_load > G1PeriodicGCSystemLoadThreshold) {
+      log_debug(gc, periodic)("Load %1.2f is higher than threshold %1.2f. Skipping.",
+                              recent_load, G1PeriodicGCSystemLoadThreshold);
+      return false;
+    }
```

Comment:
```suggestion
  if (G1PeriodicGCSystemLoadThreshold > 0.0) {
    if (os::loadavg(&recent_load, 1) == -1) {
      G1PeriodicGCSystemLoadThreshold = 0.0;
      log_warning(gc, periodic)("System loadavg() call failed, disabling G1PeriodicGCSystemLoadThreshold check.");
      // Fall through and start the periodic GC.
    } else if (recent_load > G1PeriodicGCSystemLoadThreshold) {
      log_debug(gc, periodic)("Load %1.2f is higher than threshold %1.2f. Skipping.",
                              recent_load, G1PeriodicGCSystemLoadThreshold);
      return false;
    }
```

After detecting that `loadavg` is not supported, it would be preferable (more consistent) imo that the periodic gc would be started the first time too.
Another option is to check `loadavg` support by a dummy call in argument processing, but here is good too.

As for the other change, I would prefer to have the "disabling" in active voice, making it clear that this code disables the check.

Reply:
@tschatzl  Indeed, that’s more consistent. Thank you — fixed now.